cmd/gator/validate/validate.go

```diff
@@ -0,0 +1,200 @@
+/*
+Copyright © 2021 NAME HERE <EMAIL ADDRESS>
+
+*/
+package validate
+
+import (
+	"bufio"
+	"encoding/json"
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"path/filepath"
+
+	"github.com/open-policy-agent/gatekeeper/pkg/gator/validate"
+	"github.com/spf13/cobra"
+	"gopkg.in/yaml.v2"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	k8syaml "k8s.io/apimachinery/pkg/util/yaml"
+)
+
+var (
+	errLog *log.Logger
+	outLog *log.Logger
+)
+
+const (
+	examples = `  # Validate a manifest containing Kubernetes objects, Constraint Templates, and Constraints
+  gator validate --filename="manifest.yaml"
+
+  # Validate a directory
+  gator validate --filename="config-and-policies/"
+
+  # Use multiple inputs
+  gator validate --filename="manifest.yaml" --filename="templates-and-constraints/"
+
+  # Receive input from stdin
+  cat manifest.yaml | gator validate
+
+  # Output structured violations data
+  gator validate --filename="manifest.yaml" --json`
```

Comment:
`--output=json/yaml/etc` to be more flexible and kubectl-like?

Also, should we have human-readable output?

Reply:
Default output is human readable (i.e. with no flag for a type of output).

I can change to kubectl-style.

Reply:
+1 kubectl-style

Reply:
Added support for:

- `--output=json`
- `--output=yaml`
- `-o=json`
- `-o=yaml`

Removed `--yaml` and `--json`